app/pages/verificationwithpassword/verificationwithpassword.js

```diff
@@ -0,0 +1,214 @@
+var formText2 = 'Let\'s confirm who you are and set up a password.';
+var blankLine = String.fromCharCode(160);
+
+export let VerificationWithPassword = React.createClass({
+  propTypes: {
+    acknowledgeNotification: React.PropTypes.func.isRequired,
+    api: React.PropTypes.object.isRequired,
+    notification: React.PropTypes.object,
+    signupEmail: React.PropTypes.string.isRequired,
+    signupKey: React.PropTypes.string.isRequired,
+    onSubmit: React.PropTypes.func.isRequired,
+    trackMetric: React.PropTypes.func.isRequired,
+    working: React.PropTypes.bool.isRequired
+  },
+
+  formInputs:  [
+    { type: 'explanation', text: formText }, { type: 'explanation', text: formText2 },
+    { type: 'explanation', text: blankLine },
+    { name: 'birthday', label: 'What is your date of birth?', type: 'datepicker' },
+    { type: 'explanation', text: blankLine },
+    { name: 'password', label: 'New Password', type: 'password', placeholder: '******' },
+    { name: 'passwordConfirm', label: 'Confirm password', type: 'password', placeholder: '******' }
+  ],
+
+  componentWillMount: function() {
+    this.setState({ loading: false });
+  },
+
+  getInitialState: function() {
+    var formValues = {};
+
+    return {
+      loading: true,
+      formValues: formValues,
+      validationErrors: {},
+      notification: null
+    };
+  },
+
+  isFormDisabled: function() {
+    return (this.props.fetchingUser && !this.props.user);
+  },
+
+  getSubmitButtonText: function() {
+    if (this.props.working) {
+      return 'Setting up...';
+    }
+    return 'Ready!';
+  },
+
+  render: function() {
+    return (
+      <div className="VerificationWithPassword">
+        <LoginNav
+          page="VerificationWithPassword"
+          trackMetric={this.props.trackMetric} />
+        <LoginLogo />
+        <div className="container-small-outer VerificationWithPassword-form">
+          <div className="container-small-inner VerificationWithPassword-form-box">
+            <SimpleForm
+              inputs={this.formInputs}
+              formValues={this.state.formValues}
+              validationErrors={this.state.validationErrors}
+              submitButtonText={this.getSubmitButtonText()}
+              submitDisabled={this.props.working}
+              onSubmit={this.handleSubmit}
+              notification={this.state.notification || this.props.notification} />
+          </div>
+        </div>
+      </div>
+    );
+  },
+
+  handleSubmit: function(formValues) {
+    var self = this;
+
+    if (this.props.working) {
+      return;
+    }
+
+    this.resetFormStateBeforeSubmit(formValues);
+
+    var validationErrors = this.validateFormValues(formValues);
+
+    if (!_.isEmpty(validationErrors)) {
+      return;
+    }
+
+    formValues = this.prepareFormValuesForSubmit(formValues);
+    this.props.onSubmit(this.props.api, this.props.signupKey, this.props.signupEmail, formValues.birthday, formValues.password);
```

Comment:
Huh, so the `signupEmail` is being passed in (unless it didn't get down to the component, though there should be a warning in the console if so...). Not sure why the post-success login attempt is failing on no e-mail.